gtsam/CMakeLists.txt

```diff
@@ -194,6 +196,14 @@ if(WIN32) # library to help with demangling variable names on Windows
   target_link_libraries(gtsam PRIVATE Dbghelp)
 endif()
 
+# Add warnings
+if(MSVC)
+  target_compile_options(gtsam PRIVATE /W3 /WX)
+else()
+  # No-unused-paramter isn't super helpful, as some of our functions are totally ifdefed out in some build configurations
+  target_compile_options(gtsam PRIVATE -Wall -Werror -Wpedantic -Wextra $<$<COMPILE_LANGUAGE:CXX>:-Wextra -Wno-unused-parameter>)
```

Comment:
Wow, I’m surprised this passes given that indeed some of our third libraries incur warnings.

Reply:
Looks like @Gold856 added Eigen as a system include in #1964 (see https://github.com/Gold856/gtsam/blob/bf4f6f5554c3b190539da5631e6246f146ef8fc6/cmake/HandleEigen.cmake#L45) already